pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -129,17 +129,46 @@ func buildLogsTimeSeriesFilterQuery(fs *v3.FilterSet, groupBy []v3.AttributeKey)
 	if fs != nil && len(fs.Items) != 0 {
 		for _, item := range fs.Items {
 			op := v3.FilterOperator(strings.ToLower(strings.TrimSpace(string(item.Operator))))
-			value, err := utils.ValidateAndCastValue(item.Value, item.Key.DataType)
-			if err != nil {
-				return "", fmt.Errorf("failed to validate and cast value for %s: %v", item.Key.Key, err)
+
+			var value interface{}
+			var err error
+			if op != v3.FilterOperatorExists && op != v3.FilterOperatorNotExists {
+				value, err = utils.ValidateAndCastValue(item.Value, item.Key.DataType)
+				if err != nil {
+					return "", fmt.Errorf("failed to validate and cast value for %s: %v", item.Key.Key, err)
+				}
 			}
 
 			if logsOp, ok := logOperators[op]; ok {
 				switch op {
 				case v3.FilterOperatorExists, v3.FilterOperatorNotExists:
-					columnType := getClickhouseLogsColumnType(item.Key.Type)
-					columnDataType := getClickhouseLogsColumnDataType(item.Key.DataType)
-					conditions = append(conditions, fmt.Sprintf(logsOp, columnType, columnDataType, item.Key.Key))
+					if item.Key.Type == v3.AttributeKeyTypeUnspecified {
+
+						top := "!="
+						if op == v3.FilterOperatorNotExists {
+							top = "="
+						}
+
+						if val, ok := constants.StaticFieldsLogsV3[item.Key.Key]; ok {
+							// skip for timestamp and id
+							if val.Key == "" {
+								continue
+							}
+
+							columnName := getClickhouseColumnName(item.Key)
+							if val.DataType == v3.AttributeKeyDataTypeString {
+								conditions = append(conditions, fmt.Sprintf("%s %s ''", columnName, top))
+							} else {
+								// we just have two types, number and string
+								conditions = append(conditions, fmt.Sprintf("%s %s 0", columnName, top))
+							}
+						}
+
+					} else {
+						columnType := getClickhouseLogsColumnType(item.Key.Type)
+						columnDataType := getClickhouseLogsColumnDataType(item.Key.DataType)
+						conditions = append(conditions, fmt.Sprintf(logsOp, columnType, columnDataType, item.Key.Key))
```

Comment:
Can this be changed to use the `_exists` column? I am OK if you want to address it separately.

Reply:
Right, let me add the change

Reply:
Have added the change